### PR TITLE
refactor: cache schema urls

### PR DESCRIPTION
**Pull request by richardzilincikPantheon**
Cache schema urls to /var/yang/cache/schema_dict.json. This
enables us to remove uses of `find_first_file()` and similar
functions in several places.

### CODE REVIEW

**Comment by richardzilincikPantheon** [comment]
Before trying to run `populate.py`, `generate_schema_urls.py` must be run once. Ideally this should be run on the whole `/var/yang/nonietf` directory. Running it on `standard/ietf` and `experimental/ietf-extracted-YANG-modules` might be enough to make it work.

**Comment by SlavomirMazurPantheon** [comment]
Not sure whether related to this change, but I can see following error in the logs:
```
2022-08-01 13:17:26 ERROR    modulesComplicatedAlgorithms.py modulesComplicatedAlgorithms => Existing module openconfig-aft-ethernet@2019-11-07 is missing the 'derived-semantic-version' field - 595
2022-08-01 13:17:26 ERROR    modulesComplicatedAlgorithms.py modulesComplicatedAlgorithms => Cannot resolve semver for openconfig-aft-ethernet@2017-05-10 - 597
```
If `derived-semantic-version` is not available - individual revisions should be looped and  `derived-semantic-version` should be computed

File: sandbox/generate_schema_urls.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+
+def main():
+    config = create_config()
+    cache_dir = config.get('Directory-Section', 'cache')
+    save_file_dir = config.get('Directory-Section', 'save-file-dir')
+
+    argument_parser = argparse.ArgumentParser(description=__doc__)
+    argument_parser.add_argument(
+        'directory',
```
[the remark this answers is not in the document]
Comment (richardzilincikPantheon) [reply sandbox/generate_schema_urls.py:39]: This was intentional, the `directory` argument always needs to be passed, so it's more practical to have it as a positional argument.

**Comment by richardzilincikPantheon** [comment]
I don't think the errors are related to this change. Maybe they should be addressed in a separate PR?

**Comment by SlavomirMazurPantheon** [comment]
> I don't think the errors are related to this change. Maybe they should be addressed in a separate PR?
Yes separate PR will be better, this one is already extensive.

**Pull request merged**